File: grpc-server/src/definition/entity.rs
#[derive(serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "snake_case")]
#[derive(Debug, Clone, PartialEq)]
pub struct Clan {
    pub id: u32,
    pub name: ::prost::alloc::string::String,
    pub tag: ::prost::alloc::string::String,
    pub points: u32,
    pub owner_id: ::uuid::Uuid,
    pub created_at: ::chrono::NaiveDateTime,
    pub updated_at: ::chrono::NaiveDateTime,
}
#[derive(serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "snake_case")]
#[derive(Debug, Clone, PartialEq)]
pub struct ClanMember {
    pub player_id: ::uuid::Uuid,
    pub clan_id: u32,
}
#[derive(serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "snake_case")]
#[derive(Debug, Clone, PartialEq)]
pub struct Player {
    pub uuid: ::uuid::Uuid,
    pub hexes: u32,
    pub last_username: ::prost::alloc::string::String,
    pub last_seen: ::chrono::NaiveDateTime,
    pub created_at: ::chrono::NaiveDateTime,
    pub updated_at: ::chrono::NaiveDateTime,
}
#[derive(serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "snake_case")]
#[derive(Debug, Clone, PartialEq)]
pub struct UserRole {
    pub player_id: ::uuid::Uuid,
    pub clan_id: ::prost::alloc::string::String,
}

File: kotlin-grpc-client/src/main/kotlin/org/hexalite/network/definition/entity.kt
@file:kotlinx.serialization.UseSerializers(org.hexalite.network.common.serialization.UUIDSerializer::class, org.hexalite.network.common.serialization.ChronoInstantSerializer::class)
package org.hexalite.network.definition.entity
@kotlinx.serialization.Serializable
data class Clan (
    val id: UInt,
    val name: String,
    val tag: String,
    val points: UInt,
    @kotlinx.serialization.SerialName("owner_id")
    @OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
    @kotlinx.serialization.cbor.ByteString
    val ownerId: java.util.UUID,
    @kotlinx.serialization.SerialName("created_at")
    val createdAt: kotlinx.datetime.Instant,
    @kotlinx.serialization.SerialName("updated_at")
    val updatedAt: kotlinx.datetime.Instant,
)
@kotlinx.serialization.Serializable
data class ClanMember (
    @kotlinx.serialization.SerialName("player_id")
    @OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
    @kotlinx.serialization.cbor.ByteString
    val playerId: java.util.UUID,
    @kotlinx.serialization.SerialName("clan_id")
    val clanId: UInt,
)
@kotlinx.serialization.Serializable
data class Player (
    @OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
    @kotlinx.serialization.cbor.ByteString
    val uuid: java.util.UUID,
    val hexes: UInt,
    @kotlinx.serialization.SerialName("last_username")
    val lastUsername: String,
    @kotlinx.serialization.SerialName("last_seen")
    val lastSeen: kotlinx.datetime.Instant,
    @kotlinx.serialization.SerialName("created_at")
    val createdAt: kotlinx.datetime.Instant,
    @kotlinx.serialization.SerialName("updated_at")
    val updatedAt: kotlinx.datetime.Instant,
)
@kotlinx.serialization.Serializable
data class UserRole (
    @kotlinx.serialization.SerialName("player_id")
    @OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
    @kotlinx.serialization.cbor.ByteString
    val playerId: java.util.UUID,
    @kotlinx.serialization.SerialName("clan_id")
    val clanId: String,
)
